rcon/player_history.py

```diff
@@ -275,10 +294,31 @@ def add_player_to_blacklist(steam_id_64, reason, name=None):
                     steamid=player, is_blacklisted=True, reason=reason)
             )
 
+        try:
+            if rcon and name and ban_if_in_game:
+                if name in [p['name'] for p in rcon.get_players()]:
+                    info = rcon.get_player_info(name)
+                    if info['steam_id_64'] == steam_id_64:
+                        ban_and_add_log(rcon, info['steam_id_64'], name, reason)
+        except:
+                logger.warning("Something went wrong banning player with steam id %s", steam_id_64)
+
         sess.commit()
 
 
-def remove_player_from_blacklist(steam_id_64):
+def _find_and_remove_perma_bans_by_ban_log(rcon, ban_logs, sess):
+    counter = 0
+    perma_bans = rcon.get_perma_bans()
+    for ban_log_entry in ban_logs:
```

Comment:
So in this function you could totally use the Name instead of the ban_log right? 
And as an additionally safe guard you could just chec that the time or the date roughly matches. This way you don't need the whole database layer which make the whole thing much simpler and less prone to go out of sync with the server (if for some reason the server loose its history, or if people use a different client, etc...)

Your thoughts?

Reply:
Well, one big "out of sync" problem is when the server and the RCON tool use different time zones or the time is misconfigured on the RCON tool, but this is actually not a problem when using the ban log. If the server lost its history then this function won't be called, because it is only called with actions related to the blacklist. If people use a different client, then they also pass the blacklist mechanism altogether, so I don't think it's a major issue (though we might want to set up some process to import existing bans into the blacklist/ban log).

Reply:
For the timezone issue you could add a tolerance of 24h then?

> they also pass the blacklist mechanism altogether

Fair enough